main.py

```diff
@@ -0,0 +1,223 @@
+import os
+import sys
+import datetime
+from colorama import Fore, Style
+
+LOWEST_TEMPERATURE = 'Min TemperatureC'
+HIGHEST_TEMPERATURE = 'Max TemperatureC'
+MEAN_HUMIDITY_TEMPERATURE = 'Mean Humidity'
+MAX_HUMIDITY_TEMPERATURE = 'Max Humidity'
+MONTHS = {
+    1: 'January',
+    2: 'February',
+    3: 'March',
+    4: 'April',
+    5: 'May',
+    6: 'June',
+    7: 'July',
+    8: 'August',
+    9: 'September',
+    10: 'October',
+    11: 'November',
+    12: 'December'
+
+}
+
+
+def get_files_from_drectory():
+    return os.listdir('weatherfiles/')
+
+
+def get_year_file_names(files, year):
+    year_files = []
+    for file in files:
+        if year in file:
+            year_files.append(file)
+    return year_files
+
+
+def trim_list_elements(arr):
+    stripped_elements = []
+    for element in arr:
+        stripped_elements.append(element.strip())
+    return stripped_elements
+
+
+def get_parameter_indexes(arr):
+    return dict(
+        highest_temp_index=arr.index(HIGHEST_TEMPERATURE),
+        lowest_temp_index=arr.index(LOWEST_TEMPERATURE),
+        mean_humidity_index=arr.index(MEAN_HUMIDITY_TEMPERATURE),
+        max_humidity_index=arr.index(MAX_HUMIDITY_TEMPERATURE)
+    )
+
+
+def parse_files(filenames):
+    data = list()
+    for filename in filenames:
+        file = open('weatherfiles/' + filename, 'r')
+        line = file.readline()
+        arr = line.split(',')
+        arr = trim_list_elements(arr)
+        indexes = get_parameter_indexes(arr)
+        records = list()
+        for line in file:
+            line_elements = line.split(',')
```

Comment:
Splitting on a delimiter and reading on the basis of it is not a good approach on scale. Can we explore better options of reading the weather data?

Reply:
Better ways of reading CSV with Python.